web/pdf_page_view.js

```diff
@@ -1067,6 +1067,8 @@ class PDFPageView {
         showCanvas?.(true);
         await this.#finishRenderTask(renderTask);
 
+        this.structTreeLayer ||= new StructTreeLayerBuilder(this.pdfPage);
+
```

Comment:
This seems like something that shouldn't be done unconditionally.
Also, why can't we use the local pdfPage-variable here?
```suggestion
        if (this.textLayer || this.annotationLayer) {
          this.structTreeLayer ||= new StructTreeLayerBuilder(pdfPage);
        }
```

Reply:
Also, how is overall performance affected since we'll now trigger structTree fetching/parsing *before* the textLayer is handled?

Reply:
It shouldn't change the overall performances but just the scheduling of the tasks.
Without this patch, we've:
 - get the text for the text layer
 - while getting text the struct tree is retrieved
With this patch we've:
 - get the struct tree
 - get the text for the text layer

That said if you can imagine a way to have a perf regression, please tell me.